lib/dat-worker-pool/worker.rb

```diff
@@ -49,23 +50,36 @@ def raise(*args)
     def work_loop
       @on_start_callbacks.each{ |p| p.call(self) }
       loop do
-        @on_sleep_callbacks.each{ |p| p.call(self) }
-        work_item = @queue.pop
-        @on_wakeup_callbacks.each{ |p| p.call(self) }
         break if @shutdown
-        do_work(work_item) if work_item
+        process_work
       end
     ensure
       @on_shutdown_callbacks.each{ |p| p.call(self) }
       @thread = nil
     end
 
+    def process_work
+      @on_sleep_callbacks.each{ |p| p.call(self) }
+      work_item = @queue.pop
+      @on_wakeup_callbacks.each{ |p| p.call(self) }
+      do_work(work_item) if work_item
+    rescue StandardError => exception
+      handle_exception(exception, work_item)
+    rescue ShutdownError => exception
+      handle_exception(exception, work_item)
+      raise exception
```

Comment:
@jcredding can we be paranoid and put the shutdown error rescue above std err?  Just in case shutdown err ever changes to become a std err.